code/Helper/Entity/Producthelper.php

```diff
@@ -812,17 +812,17 @@ public function getObject(Mage_Catalog_Model_Product $product)
                         $customData[$attribute_name] = array_values(array_unique($values));
                     }
 
+                    // Set main product out of stock if all
+                    // sub-products are out of stock.
                     if ($customData['in_stock'] && $all_sub_products_out_of_stock) {
-                        // Set main product out of stock if all
-                        // sub-products is out of stock.
                         $customData['in_stock'] = 0;
                     }
-                } else {
+                } elseif ($value === null) {
                     $value = $this->getValueOrValueText($product, $attribute_name, $attribute_resource);
+                }
 
-                    if ($value) {
-                        $customData[$attribute_name] = $value;
-                    }
+                if ($value && !isset($customData[$attribute_name])) {
```

Comment:
@maxiloc I didn't add is as else because it would mean some duplicated code once we would have more `elseif`.
But I'm not sure about that. Mainly because `isset`. Is it intentional that the attribute already exists in `$customData` and is in `$additionalAttributes` to be overridden again?